fetch-merchant public

jayesh1602 opened the pull request:
review and merge

jayesh1602 committed:
fetch-merchant public

rohitvpatil0810 commented:
@ManasJadhav Please review this MR.

ManasJadhav commented on src/main/java/com/whytowait/api/v1/controllers/FetchMerchantInfoController.java:
```diff
@@ -0,0 +1,33 @@
+    FetchMerchantInfoService fetchMerchantInfoService;
+
+    @GetMapping(path = "/merchant")
+    public SuccessResponse<FetchBasicAndTimingResDTO> fetchMerchantInfo(@RequestParam(name = "query") String query , @RequestBody FetchMerchantInfoReqDTO requestDTO) {
```
why don't we take merchantId from the Param itself?

ManasJadhav commented on src/main/java/com/whytowait/api/v1/controllers/FetchMerchantInfoController.java:
```diff
@@ -0,0 +1,33 @@
+import java.util.List;
+
+@RestController
+@RequestMapping("/fetch")
```
use restAPI structure.
ex - merchants/:merchantId

ManasJadhav commented on src/main/java/com/whytowait/api/v1/controllers/FetchMerchantInfoController.java:
```diff
@@ -0,0 +1,33 @@
+
+        FetchBasicAndTimingResDTO fetchBasicInfoDTO = fetchMerchantInfoService.getMerchantUserDetails(requestDTO.getMid(), query);
+        String responseMessage ="Merchant Data Fetched";
+        if(fetchBasicInfoDTO==null){
```
if not found throw notFoundException from service only.

ManasJadhav commented on src/main/java/com/whytowait/api/v1/services/FetchMerchantInfoService.java:
```diff
@@ -0,0 +1,71 @@
+public class FetchMerchantInfoService {
+
+    @Autowired
+    FetchMerchantInfoRepository fetchMerchantInfoRepository;
```
move this to merchant service,  no need for different service

rohitvpatil0810 replied:
Separated due to public API, renaming to Public prefix files and classes.

ManasJadhav commented on src/main/java/com/whytowait/domain/dto/fetchMerchantInfo/FetchMerchantInfoReqDTO.java:
```diff
@@ -0,0 +1,10 @@
+import java.util.UUID;
+
+@Data
+public class FetchMerchantInfoReqDTO {
```
No need, if we take from param or query itself.

ManasJadhav commented on src/main/java/com/whytowait/domain/dto/fetchMerchantInfo/FetchBasicInfoDTO.java:
```diff
@@ -0,0 +1,28 @@
+
+@Data
+
+public class FetchBasicInfoDTO {
```
No need for a separate DTO for information and timings, you can handle that in one DTO only.
Jackson will take care of nulls.

ManasJadhav commented on src/main/java/com/whytowait/domain/dto/fetchMerchantInfo/FetchTimingProjection.java:
```diff
@@ -0,0 +1,10 @@
+
+import java.time.LocalTime;
+
+public interface FetchTimingProjection {
```
use of this ??

rohitvpatil0810 replied:
removed changed to jdbcTemplate

ManasJadhav commented on src/main/java/com/whytowait/repository/FetchMerchantInfoRepository.java:
```diff
@@ -0,0 +1,38 @@
+import java.util.UUID;
+
+@Repository
+public interface FetchMerchantInfoRepository extends JpaRepository<Merchant, UUID> {
```
Move this to merchantRepo, no need for different repo

rohitvpatil0810 replied:
Separate Repo due to JdbcTemplate implementation and public exposed API .

ManasJadhav commented on src/main/java/com/whytowait/api/v1/services/FetchMerchantInfoService.java:
```diff
@@ -0,0 +1,71 @@
+
+    public FetchBasicAndTimingResDTO getMerchantUserDetails(UUID mid, String param) {
+        FetchBasicAndTimingResDTO resDTO = new FetchBasicAndTimingResDTO();
+        if(param.equalsIgnoreCase("basic")){
```
can we have a single enum for param inputs
like a basic, basic address.

This will help in reading the code.

jayesh1602 committed:
update and fetch merchant formatting

rohitvpatil0810 commented on src/main/java/com/whytowait/api/v1/controllers/FetchMerchantInfoController.java:
Change Name of this controller - public endpoints package -> merchant -> merchantController

rohitvpatil0810 commented on src/main/java/com/whytowait/api/common/enums/ParamType.java:
change file name to MerchantInfoQueryParamsType

rohitvpatil0810 commented on src/main/java/com/whytowait/api/v1/controllers/FetchMerchantInfoController.java:
```diff
@@ -1,33 +1,25 @@
 
 @RestController
-@RequestMapping("/fetch")
+@RequestMapping(path = "/get")
```
change path to /public

rohitvpatil0810 commented on src/main/java/com/whytowait/api/v1/controllers/MerchantController.java:
```diff
@@ -27,4 +31,22 @@ public ApiResponse<CreateMerchantResponseDTO> createMerchant(@Valid @RequestBody
         return new SuccessResponse<CreateMerchantResponseDTO>("Merchant Created Successfully", CreateMerchantResponseDTO.fromMerchant(response));
     }
+
+    @PostMapping(path = "/update-address")
```
Create new controller AddressController - PUT method to update address.

rohitvpatil0810 commented on src/main/java/com/whytowait/api/v1/controllers/MerchantController.java:
```diff
@@ -27,4 +31,22 @@ public ApiResponse<CreateMerchantResponseDTO> createMerchant(@Valid @RequestBody
+    public ApiResponse<Address> updateMerchant(@Valid @RequestBody UpdateMerchantAddressReqDTO requestDTO) throws BadRequestException {
+        Address  response = merchantCreateService.updateMerchantAddress(requestDTO);
+        if(response==null){
+            return new SuccessResponse<Address>("Merchant Not Found/Exception Occured");
```
Should use notFound or badRequestError Exception globalexception manager will handle it.

rohitvpatil0810 commented on src/main/java/com/whytowait/api/v1/services/FetchMerchantInfoService.java:
Put the similar structure like controller -> public -> merchant -> merchantService. no need to add fetch in front of file name or service name.

rohitvpatil0810 commented on src/main/java/com/whytowait/api/v1/services/FetchMerchantInfoService.java:
```diff
@@ -1,71 +1,58 @@
+        FetchMerchantInfoResDTO resDTO = new FetchMerchantInfoResDTO();
+        ParamType paramType = ParamType.fromString(param);
+
+        if(paramType == ParamType.BASIC){
```
Use the switch on the type of query param and create a functions for handling each type.

rohitvpatil0810 commented on src/main/java/com/whytowait/api/v1/services/MerchantService.java:
```diff
@@ -56,4 +65,43 @@ public Merchant createMerchant(CreateMerchantRequestDTO createMerchantReqDTO) th
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        UserDetailsDTO userDetails = (UserDetailsDTO) authentication.getPrincipal();
+        User user = userRepository.findByMobile(userDetails.getMobile());
+        Merchant fetchMerchantByOwnerId = merchantRepository.findByOwnerId(user.getId());
```
rename variable name to merchant - function call is sufficient for what is understanding what is happening

rohitvpatil0810 replied:
add checkMerchantRole annotation to verify user authrization to update merchant address, put required as owner.

rohitvpatil0810 commented on src/main/java/com/whytowait/api/v1/services/MerchantService.java:
```diff
@@ -56,4 +65,43 @@ public Merchant createMerchant(CreateMerchantRequestDTO createMerchantReqDTO) th
+        return  null;
+    }
+
+    public User updateMerchantBasicInfo(UpdateMerchantDetailReqDTO reqDTO) throws BadRequestException{
```
the function name is updateMerchantBasicInfo but updating user. what is happening here?

rohitvpatil0810 replied:
function should update merchant details like restaurantName, etc. not user name owning restaurant, that will happened in usercontroller.

rohitvpatil0810 replied:
add checkMerchantRole annotation to verify user authrization to update merchant details, put required as owner.

rohitvpatil0810 commented on src/main/java/com/whytowait/domain/dto/fetchMerchantInfo/FetchTimingsDTO.java:
no content in file - remove this file.

rohitvpatil0810 commented on src/main/java/com/whytowait/domain/dto/merchant/UpdateMerchantDetailReqDTO.java:
This should be changed with Merchant Info fields not user details fields.

rohitvpatil0810 commented on src/main/java/com/whytowait/domain/dto/fetchMerchantInfo/FetchAddressInfoDTO.java:
no content in file, remove the file

rohitvpatil0810 commented on src/main/java/com/whytowait/domain/dto/fetchMerchantInfo/TimingDTO.java:
fi not required remove this commented code.

jayesh1602 committed:
updation formatting

ManasJadhav commented:
Why do we need different service and repo like fetchMerhant....
Can't this be done in merchant only

rohitvpatil0810 commented:
I am taking this PR changes to my local and will push updated code.

rohitvpatil0810 commented:
I am taking this PR changes to my local and will push updated code.

rohitvpatil0810 commented:
I am taking this PR changes to my local and will push updated code.

rohitvpatil0810 committed:
chore: Merge branch 'main' into feature/fetch-merchant

rohitvpatil0810 committed:
fix: fix CheckMerchantRoleAspect to get path variable

rohitvpatil0810 committed:
chore: refactor code
fix: add merchantId in path variable for update merchant and update address route - change method PUT
chore: rename FetchMerchantInfoController.java,
 FetchMerchantInfoService.java, FetchMerchantInfoRepo.java to PublicMerchantInfoController.java, PublicMerchantInfoService.java and PublicMerchantInfoRepo.java

rohitvpatil0810 commented:
@ManasJadhav Review this PR.
